### ISSUE TITLE
Custom functions with no return possible?

### ISSUE THREAD

**Issue by h1ps** [josdejong/mathjs#776]
Hi, 
is it possible to create an own function like
```javascript
function extract(param1, param2){
//do something with param1/2
   return ???
}
extract.toTex = function (node, options) {
math.import({extract: extract});
```
that returns "nothing". 
If i return _null_ or _undefined_ or "" (empty string) math.js interpret that value, but i want something that is unkown or hidden for math.js, so that is not disturbing the actually calculation

 Thank you in advance!

**Comment by josdejong** [comment]
Yes you can always leave out a return statement, in that case JavaScript functions return `undefined`:

```js
function extract(param1, param2){
  console.log(param1, param2)
}
math.import({extract: extract})

var res = math.extract(2,3)
// now res === undefined
// and console output is "2 3"
```

**Comment by h1ps** [comment]
Hi Jos, thanks for your reply. 
Hmm, that works in case if i have nothing else in my expression. 

I will try to rephrase what i mean.
An other example. In my input i am looking for something like 
```javascript
test(x) = x^2 extract(x) 
test(2)  //should be 4
```
It is understandable that this line throws an error because math.js think it is a multiply. Thats the reason why i ask for a "hidden" return type. 

I hope its clearer what i mean :>

**Comment by josdejong** [comment]
Are you looking for some kind of a hook which is invoked every time a created function is executed, maybe for logging purposes or something? Can you explain your use case?

A trick could be:

```
test(x) = [x^2, extract(x)][1]
test(2)  // 4
```

**Comment by h1ps** [comment]
My use case is, that auser has a code editor for mathematical calculations. In some context it is necessary to exctract and plot some values in a diagram. And it would be nice if the extract function does not influence the actual calculation.

**Comment by josdejong** [comment]
So the `extract(x)` is entered by the user when the user wants extra context? Or is this used behind the scenes?

**Comment by h1ps** [comment]
yes, it would be great if it could be entered by the user

**Comment by josdejong** [comment]
Ok I don't think that's doable (except for using the trick that I mentioned before). What's missing here is multi-line support in defining functions, that would solve it because then you can first have a line `extract(x)` and then `return x^2`. Right now function definitions are just a single expression, not a block of code.

Maybe an option is to have a checkbox "extract" right from the input field when the user entered a function definition, and in that case you could wrap the function in another function which would extract all input variables, execute the defined function, and return it's results.
